src/main/java/gregtech/api/pipenet/block/BlockPipe.java

```diff
@@ -656,18 +656,26 @@ private List<IndexedCuboid6> getCollisionBox(IBlockAccess world, BlockPos pos, @
 
     public boolean hasPipeCollisionChangingItem(IBlockAccess world, BlockPos pos, Entity entity) {
         if (entity instanceof EntityPlayer) {
-            return hasPipeCollisionChangingItem(world, pos, ((EntityPlayer) entity).getHeldItem(EnumHand.MAIN_HAND)) ||
-                    hasPipeCollisionChangingItem(world, pos, ((EntityPlayer) entity).getHeldItem(EnumHand.OFF_HAND)) ||
+            return hasPipeCollisionChangingItem(world, pos, ((EntityPlayer) entity).getHeldItem(EnumHand.MAIN_HAND),
+                    entity) ||
+                    hasPipeCollisionChangingItem(world, pos, ((EntityPlayer) entity).getHeldItem(EnumHand.OFF_HAND),
+                            entity) ||
                     entity.isSneaking() && isHoldingPipe((EntityPlayer) entity);
         }
         return false;
     }
 
     public abstract boolean isHoldingPipe(EntityPlayer player);
 
-    public boolean hasPipeCollisionChangingItem(IBlockAccess world, BlockPos pos, ItemStack stack) {
+    public boolean hasPipeCollisionChangingItem(IBlockAccess world, BlockPos pos, ItemStack stack,
+                                                @Nullable Entity entity) {
         if (isPipeTool(stack)) return true;
 
+        if (entity != null && entity.isSneaking() && entity instanceof EntityPlayer player &&
+                (GTUtility.isSprayCan(player.getHeldItem(EnumHand.MAIN_HAND)) ||
+                        GTUtility.isSprayCan(player.getHeldItem(EnumHand.OFF_HAND))))
+            return true;
```

Comment:
This should check `stack` instead of retrieving the held item in both hands. This method will already be called twice, once for each hand, and in specific cases only called for one of the hands on purpose. Therefore it should not check both hands when it is intended that only one hand is checked per call.

As a result, `entity` no longer needs to be a parameter so it should be removed as well.

Reply:
the entity parameter is needed for the crouch check, however this is unnecessary and is a holdover from the previous method